klient/k8s/resource/resource.go

```diff
@@ -41,11 +41,11 @@ type Resources struct {
 	client cr.Client
 }
 
-// Res instantiates the controller runtime client
+// NewResource instantiates the controller runtime client
 // object. User can get panic for belopw scenarios.
 // 1. if user does not provide k8s config
```

Comment:
As a follow-up, this should probably return an error and the tests can be updated to check the error instead of capturing a panic

Reply:
I think if initialisation of controller runtime client fails, the flow should be terminated as without client there is no operation that can be performed!